compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/FirDataFlowAnalyzer.kt

```diff
@@ -833,7 +844,7 @@ class FirDataFlowAnalyzer(private val components: FirAbstractBodyResolveTransfor
         if (fir is FirThisReceiverExpressionImpl) {
             return variableStorage.getOrCreateNewThisRealVariable(fir.calleeReference.boundSymbol ?: return null)
         }
-        val symbol: AbstractFirBasedSymbol<*> = fir.resolvedSymbol ?: return null
+        val symbol = fir.resolvedSymbol as? FirVariableSymbol<*> ?: return null
```

Comment:
I made this change because of an issue I found with `codegen/box/unaryOp/callNullable.kt` (snippet below):

```kt
    val a1: Byte? = 1.unaryMinus()
    val a2: Short? = 1.unaryMinus()
    val a3: Int? = 1.unaryMinus()
// ...
    if (a1!! != (-1).toByte()) return "fail 1"
    if (a2!! != (-1).toShort()) return "fail 2"
    if (a3!! != -1) return "fail 3"
```

`exitVariableDeclaration` would create a "real variable" for the initializer with the `unaryMinus` function as the symbol, and create an alias variable for `a1` pointing to the `unaryMinus` variable. It would also create similar alias variables for `a2`, `a3` etc. Each `!!` call would add the not-nullable type to the type info for the `unaryMinus` function, which does not make sense (it builds an intersection type, e.g., `Byte & Short & Int` after `a3!!`).

FWIW, this issue also exists with the `as` type operator. If you changed all the `!!` to `as` (e.g., `a1 as Byte` instead of `a1!!`, etc.), you would see some failure: `Caused by: java.lang.ClassCastException: java.lang.Short cannot be cast to java.lang.Byte` (JVM tries to cast `a2` to `Byte`).

This change makes sure that only put real variable symbols in storage. Let me know if this seems right (or not)!

Reply:
Yeah, it seems good

Actually, there is a bunch of bugs with aliased variables and I am planning to rewrite all code that is relative to them (and, probably, get rid of alias variables at all)